### PR TITLE
Update Open and Loading

### PR DESCRIPTION
**Pull request by mikedijkstra**
The changes in this should hopefully reduce any issues found with having multiple LiveChatLoader providers in your app and improve loading transitions between the fake widget and the real provider widget.

This PR will:

- Remove unused state checks from the loadChat hook (this function is wrapped in a useCallback so the values don't change)
- Update how we verify if a provider has been asked to load. This logic is know within the provider itself
- Update how we check if a provider has fully loaded
- Update individual providers to use APIs where appropriate
- Update build so we use the version built in the repo on the website to test updates as they're in progress
- Update Intercom close state

### CODE REVIEW

File: src/providers/drift.ts [hunk 1]
```diff
@@ -68,23 +62,29 @@ const loadScript = () => {
...
-  window.drift.on('ready', () => {
-    setState('open')
-    setTimeout(() => setState('complete'), 2000)
-  })
+}): boolean => {
+  const loaded = loadScript()
+
+  if (loaded) {
+    window.drift.load(providerKey)
+    window.drift.SNIPPET_VERSION = '0.3.1'
```
Comment (benschwarz) [inline src/providers/drift.ts:81]: Does this lock the drift version? Is it worth a bump to current at the same time?

File: src/providers/intercom.ts [hunk 2]
```diff
@@ -42,24 +41,34 @@ const loadScript = () => {
...
   providerKey: string
   setState: (state: State) => void
-}) => {
-  loadScript()
-  window.Intercom('boot', { app_id: providerKey })
-  setTimeout(() => setState('complete'), 2000)
+}): boolean => {
+  const loaded = loadScript()
+
+  if (loaded) {
```
Comment (benschwarz) [inline src/providers/intercom.ts:57]:
```suggestion
  // Continue as long as Intercom hasn’t already been initialised. 
  if (loaded) {
```

?

File: src/utils/waitForLoad.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    if (check() || elapsedTime >= maxWaitDuration) {
+      callback()
+      return
+    }
+
+    elapsedTime += deadline.timeRemaining()
+    requestIdleCallback?.(scheduleLoad)
+  }
+
+  if (requestIdleCallback) {
```
Comment (benschwarz) [inline src/utils/waitForLoad.ts:23]: Should this be `window.requestIdleCallback` for clarity?

Comment (benschwarz) [reply src/utils/waitForLoad.ts:23]: Ah. No. I re-read the first line. Disregard ⚡

**Review by benschwarz — APPROVED** [review]
LGTM

File: src/providers/drift.ts (hunk 1 above)
Comment (mikedijkstra) [reply src/providers/drift.ts:82]: I'm not sure what the current is 🤔

**Comment by mxstbr** [comment]
Yesss, this is amazing — I can't wait for this to land! :tada: Might also be worth rolling this fix into the release: https://github.com/calibreapp/react-live-chat-loader/pull/76

**Pull request merged**